Disable broken and outdated CI

tobiasdiez opened the pull request:
<!-- ^ Please provide a concise and informative title. -->
<!-- ^ Don't put issue numbers in the title, do this in the PR description below. -->
<!-- ^ For example, instead of "Fixes #12345" use "Introduce new method to calculate 1 + 2". -->
<!-- v Describe your changes below in detail. -->
<!-- v Why is this change required? What problem does it solve? -->
<!-- v If this PR resolves an open issue, please link to it here. For example, "Fixes #12345". -->

Many of the CI runs after a new release are failing, for months now, see eg https://github.com/sagemath/sage/actions/runs/12979684199/job/36218126145. Some of these failures are genuine (eg a certain package cannot be built on a certain system) and some others are due to constraints of the build system (eg running out of free space). Since there is very little point in senselessly burning energy, all runs that were failing for the last releases are disabled. Once the underlying issues are fixed, they can be easily be re-enabled.

Moreover, the "minimal" runs where only a couple of systems packages are installed and most are build using sage are removed, keeping only the "standard" where all available system packages (corresponding to standard sage packages) are installed. 

New test run: https://github.com/tobiasdiez/sage/actions/runs/13199372232/job/36847711005

### :memo: Checklist

<!-- Put an `x` in all the boxes that apply. -->

- [ ] The title is concise and informative.
- [ ] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation and checked the documentation preview.

### :hourglass: Dependencies

- #39251 (to resolve merge conflicts)

<!-- List all open PRs that this PR logically depends on. For example, -->
<!-- - #12345: short description why this is a dependency -->
<!-- - #34567: ... -->

tobiasdiez committed:
Disable broken and outdated CI

tobiasdiez committed:
Fail ci-retrofit if it's not yet configured

tobiasdiez committed:
Remove obsolete "if" clause in CI

tobiasdiez committed:
Revert "Fail ci-retrofit if it's not yet configured"

This reverts commit f93a2d77ed83c7fa7910e6ac5fb6e8d379ce937b.

tobiasdiez committed:
add targets_optional: ptest

tobiasdiez committed:
Remove failing retrofit

tobiasdiez committed:
Remove more failing systems

kwankyu commented on .github/workflows/ci-linux.yml:
```diff
@@ -33,157 +32,15 @@ permissions:
-        ["standard"]
-      docker_push_repository: ghcr.io/${{ github.repository }}/
-      logs_artifact: false
-
```
Why delete this? What is broken?

kwankyu commented on .github/workflows/ci-linux.yml:
```diff
@@ -33,157 +32,15 @@ permissions:
-      # Make sure that all "standard" jobs can start simultaneously,
-      # so that runners are available by the time that "default" starts.
-      max_parallel: 50
-
```
Why delete this? What is broken?

user202729 commented on .ci/write-dockerfile.sh:
```diff
@@ -349,14 +349,9 @@ ENV SAGE_CHECK_PACKAGES="!cython,!python3,!cysignals,!linbox,!ppl,!cmake,!rpy2,!
-    fi; \\
+    rm -rf /sage/src; \\
+    mv src /sage/src; \\
+    cd /sage && ./bootstrap && ./config.status; \\
```
By the way what is this part supposed to do? As I read retrofit-worktree does

```
    echo >&2 "usage: $0 WORKTREE_NAME WORKTREE_DIRECTORY"
    echo >&2 "Ensures that the current working directory is a git repository,"
    echo >&2 "then makes WORKTREE_DIRECTORY a git worktree named WORKTREE_NAME."
```

but I don't understand why. Your commit would make sense if you assume the retrofit-worktree always fail (but why would it?)

kwankyu replied:
retrofit-worktree is to perform "incremental build" from the previous stage.

The change here makes ""incremental build" broken, and thus also the whole point of the multi-stage docker builds.

By the way, I made changes to simplify the multi-stage docker builds in the other PR.

kwankyu commented on .github/workflows/ci-linux.yml:
```diff
@@ -33,157 +32,15 @@ permissions:
-      max_parallel: 8
-
-  minimal:
-    if: ${{ success() || failure() }}
```
Why delete this? 

There are broken platforms. We may selectively turn them off until fixed.

dimpase replied:
"gentoo-python3.10" is dead, as Gentoo doesn't ship python 3.10 any more (and 3.11 will soon be gone, too)

tobiasdiez replied:
> Why delete this?

See PR description.

kwankyu commented on .github/workflows/ci-linux.yml:
```diff
@@ -33,157 +32,15 @@ permissions:
-  maximal-pre:
-    if: ${{ success() || failure() }}
-    needs: [minimal]
-    uses: ./.github/workflows/docker.yml
```
Why delete this?

There are broken platforms. We may selectively turn them off until fixed.

tobiasdiez replied:
> Why delete this?

This is not deleted, only renamed to `maximal`.

kwankyu commented on .github/workflows/ci-linux.yml:
```diff
@@ -33,157 +32,15 @@ permissions:
-
-  optional:
-    if: ${{ success() || failure() }}
-    needs: [maximal-pre]
```
Why delete this?

We may turn this off until fixed.

tobiasdiez replied:
> Why delete this?
> 
> We may turn this off until fixed.

You cannot turn single steps off as far as I know. And all of them are currently failing.

kwankyu commented on .github/workflows/ci-linux.yml:
```diff
@@ -33,157 +32,15 @@ permissions:
-      targets_optional: '$(echo $(export PATH=build/bin:$PATH && sage-package list :optional: --has-file "spkg-install.in|spkg-install|requirements.txt" --no-file "huge|has_nonfree_dependencies" | grep -v sagemath_doc | grep -v ^_))'
-
-  experimental:
-    if: ${{ success() || failure() }}
```
Why delete this?

We may turn this off until fixed.

tobiasdiez replied:
see above

kwankyu commented on .github/workflows/ci-macos.yml:
```diff
@@ -99,61 +99,3 @@ jobs:
-
-    needs: [dist]
-
-    runs-on: ${{ matrix.os }}
```
Is this also failing? I cannot find a run.

tobiasdiez replied:
Yes, all of them are failing see eg https://github.com/sagemath/sage/actions/runs/12979684206/job/36196002577

kwankyu commented on .github/workflows/docker.yml:
```diff
@@ -50,18 +43,11 @@ on:
-           "debian-bullseye-i386",
           ]
         # 'tox -e update_docker_platforms' updates above
       tox_packages_factors:
```
See the comment at L52. The main list is in `tox.ini`.

tobiasdiez replied:
I only changed the ci flows on purpose so that people trying to fix these other systems have an easy way to do this via tox/dev container. But I can completely remove them everywhere if you don't think that anyone will actively work on them.

kwankyu replied:
> I only changed the ci flows on purpose so that people trying to fix these other systems have an easy way to do this via tox/dev container. 

Do you know that removing platforms from the CI workflow also removes the corresponding docker images?

> But I can completely remove them everywhere if you don't think that anyone will actively work on them.

You and I don't know who does and will work on what platforms. The default list of platforms tested in CI should be determined by what platforms we want to support.

dimpase replied:
Anything that's not documented (in Sage documentation) is subject to change without notice, and our documentation does not promise anyone a million docker images ready to use. 


Thus "you and I don't know how these images are used" is not an argument to preserve them.

kwankyu replied:
I doubt whether you (and the author as well) ever read through https://doc-release--sagemath.netlify.app/html/en/developer/portability_testing.

dimpase replied:
> I doubt whether you (and the author as well) ever read through https://doc-release--sagemath.netlify.app/html/en/developer/portability_testing.

of course I read it, I used it, etc. Please don't give such dismissive mkoeppe-style responses.

The concrete names of containers are mentioned there in *examples*. 
These are not cut in stone, they are just mentioned as examples. Just as examples might mention concrete, random, versions, random dates, etc, random outputs of computations, etc. This is not what documenting means.

kwankyu replied:
Did you see this section: https://doc-release--sagemath.netlify.app/html/en/developer/portability_testing#using-our-pre-built-docker-images-published-on-ghcr-io ?

That is the documentation of available docker images.

The proper way to update the documentation is what I mentioned in the beginning of this thread: https://github.com/sagemath/sage/pull/39467#discussion_r1947540858

dimpase replied:
it's pretty clear that it's a collection of semi- random docker images, some of them quite old, corresponding to rather random betas, etc. 

It's too ambitious to support even half of them, too much work that noone would be doing

kwankyu replied:
There is the master list of the linux platforms we support in `tox.ini` (in the sense of @user202729 "we test it and fix it if it is broken") 

I don't care what you think (and the author as well) should be in the list. Just update it properly by `tox -e update_docker_platforms` as the comment at Line 52 says. If not done properly, our documentation is broken. #39009 does it properly.

In the same vein, removing "minimal" and "optional" and "maximal" jobs is not a proper way to fix CI Linux. We have to support sage-the-distribution, as in the documentation.

tobiasdiez replied:
There is value in having this table, even if the CI currently doesn't build new images for all listed configs. You clearly see the latest release it was working.

Moreover, removing it from the tox.ini also removes the codespaces, which is the easiest way to actually work on these images. So if anyone want's to work on fixing the issues, then we should give people the tools to do so. This PR is not about removing support for a given system, just removing failing CI runs.

tobiasdiez committed:
Use maximal in place of standard for CI

tobiasdiez committed:
Clarify portability testing docs

tobiasdiez committed:
Merge remote-tracking branch 'upstream/develop' into disable-broken-ci

tobiasdiez committed:
Remove fedora systems after EOL

tobiasdiez committed:
Readd ubuntu-jammy-minimal

tobiasdiez committed:
Merge remote-tracking branch 'origin/py-upgrade' into disable-broken-ci

tobiasdiez committed:
Use 'standard' instead of 'maximal' as package factor

'maximal' also installs optional packages as system packages; this is better handled via the 'optional' factor.

kwankyu commented on .github/workflows/ci-linux-incremental.yml:
```diff
@@ -99,56 +99,9 @@ jobs:
       max_parallel: 8
-
-  constraints_pkgs-norequirements:
-    needs: [changed_files]
```
Please refrain from removing things just because it is broken. They can be fixed later. Turning off is enough.

kwankyu commented on .github/workflows/ci-linux.yml:
```diff
@@ -66,124 +43,21 @@ jobs:
       targets_optional: ptest
+      tox_system_factors: >-
+          ["ubuntu-jammy"]
       tox_packages_factors: >-
```
At the end, you renamed "default" to "minimal", effectively removing "minimal" configurations. 

The original "minimal" configurations serve a purpose, as discussed in the other PR.

kwankyu replied:
The "minimal" configurations are intended to test standard sage packages.

kwankyu commented on .github/workflows/ci-linux.yml:
```diff
@@ -66,124 +43,21 @@ jobs:
-      max_parallel: 50
-
-  standard-constraints_pkgs-norequirements:
-    if: ${{ success() || failure() }}
```
Just turn this off. Please do not remove things broken. They may be fixed later.

kwankyu commented on .github/workflows/ci-linux.yml:
```diff
@@ -66,124 +43,21 @@ jobs:
-      max_parallel: 15
 
-  standard-sitepackages:
-    if: ${{ success() || failure() }}
```
Just turn this off. Please do not remove things broken. They may be fixed later.

kwankyu commented on .github/workflows/ci-linux.yml:
```diff
@@ -66,124 +43,21 @@ jobs:
-      max_parallel: 24
-
-  maximal-pre:
-    if: ${{ success() || failure() }}
```
This is intended to test "optional packages" installed as system package.

kwankyu commented on .github/workflows/ci-linux.yml:
```diff
@@ -66,124 +43,21 @@ jobs:
       docker_push_repository: ghcr.io/${{ github.repository }}/
-
-  optional:
-    if: ${{ success() || failure() }}
```
This is intended to test "optional sage packages"

kwankyu commented on .github/workflows/ci-linux.yml:
```diff
@@ -66,124 +43,21 @@ jobs:
-      targets_optional: '$(echo $(export PATH=build/bin:$PATH && sage-package list :optional: --has-file "spkg-install.in|spkg-install|requirements.txt" --no-file "huge|has_nonfree_dependencies" | grep -v sagemath_doc | grep -v ^_))'
-
-  experimental:
-    if: ${{ success() || failure() }}
```
This is intended to test "experimental sage packages"

tobiasdiez committed:
Merge branch 'develop' into disable-broken-ci

tobiasdiez committed:
Restore retrofit
